install_sct

```diff
@@ -259,28 +257,53 @@ function get_shell_rc_path() {
   fi
 }
 
-# Force bashrc loading
-function force_bashrc_loading() {
-  sourceblock="
-if [[ -n \"\$BASH_VERSION\" ]]; then
-    # include .bashrc if it exists
-    if [[ -f \"\$HOME/.bashrc\" ]]; then
-    . \"\$HOME/.bashrc\"
-    fi
-fi"
-  bidon=""
-  for profile in ~/.bash_profile ~/.bash_login ~/.profile; do
-    if [[ -a "$profile" ]]; then
-      if ! grep -E "(\.|source) .*bashrc" "$profile" >/dev/null 2>&1; then
-        echo "$sourceblock" >>"$profile"
-      fi
-      bidon="done"
-      break
-    fi
-  done
+# Deal with permission issues on files used by SCT installer
+# It is strangely common for a user to not have read/write access to files
+# present in their own home directory (e.g. not owning .bashrc, somehow).
+# So, we provide advice here so that they can hopefully fix the issue
+# themselves, rather than having to contact us.
+function check_permissions() {
+  FILE_TO_CHECK="$1"
+  echo -n "Checking permissions on required file $FILE_TO_CHECK:"
+
+  if [ -e "$FILE_TO_CHECK" ]; then EXIST="True"; else EXIST="False"; fi
+  if [ -O "$FILE_TO_CHECK" ]; then OWNED="True"; else OWNED="False"; fi
+  if [ -r "$FILE_TO_CHECK" ]; then READS="True"; else READS="False"; fi
+  if [ -w "$FILE_TO_CHECK" ]; then WRITE="True"; else WRITE="False"; fi
+
+  MSG_EXIST="  - File exists: $EXIST"
+  MSG_OWNED="  - File ownership: $OWNED"
+  MSG_READS="  - Read permissions: $READS"
+  MSG_WRITE="  - Write permissions: $WRITE"
+
+  WARNING="False"
+  if [ $EXIST == "False" ]; then
+    MSG_EXIST+="\n    NOTE: The SCT installer will create this file."
+  fi
+  if [ $OWNED == "False" ]; then
+    MSG_OWNED+="\n    WARNING: If you encounter permission issues, you can try running the following command:"
+    MSG_OWNED+="\n             sudo chown \"$(whoami)\" \"$FILE_TO_CHECK\""
+    WARNING="True"
+  fi
+  if [ $READS == "False" ]; then
+    MSG_READS+="\n    WARNING: If you encounter permission issues, you can try running the following command:"
+    MSG_READS+="\n             sudo chmod u+r \"$FILE_TO_CHECK\""
+    WARNING="True"
+  fi
+  if [ $WRITE == "False" ]; then
+    MSG_WRITE+="\n    WARNING: If you encounter permission issues, you can try running the following command:"
+    MSG_WRITE+="\n             sudo chmod u+w \"$FILE_TO_CHECK\""
+    WARNING="True"
+  fi
 
-  if [[ -z "$bidon" ]]; then
-    echo "$sourceblock" >>~/.bash_profile
+  # Note: Ideally we would properly color each individual message, however `print` adds a *lot* of
+  #       unnecessary spacing, so chaining multiple `print` messages looks very bad. We could remove
+  #       the spacing from `print`, but this would drastically change the output of our installer.
+  MSG_TO_WRITE="$MSG_EXIST\n$MSG_OWNED\n$MSG_READS\n$MSG_WRITE"
+  if [ $WARNING == "True" ]; then
+    print warning "$MSG_TO_WRITE"
+  else
+    print info "$MSG_TO_WRITE"
   fi
```

Comment:
Note that this is how the message currently looks:

![image](https://github.com/user-attachments/assets/80ec471b-b2c3-4efa-b26e-2175bf33f938)

And for e.g. an unreadable .bashrc:

![image](https://github.com/user-attachments/assets/e38482bd-c4fb-402f-b613-1339fac0d015)

I would prefer to have more granular coloring, but the `print` function we created for `install_sct` is a bit finicky (see above comment).